server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -464,6 +466,25 @@ private int submitCompactionTasks(
         }
       }
 
+      CompactionEngine compactionEngine = config.getEngine();
+      Map<String, Object> autoCompactionContext = newAutoCompactionContext(config.getTaskContext());
+      int numCurrentCompactionTasksAndSubtasks;
+      final String maxNumTasksContextParam = "maxNumTasks";
```

Comment:
This should be a constant in the `MSQContext` class in `ClientCompactionRunnerInfo`.
Also, the `MSQContext` should be a class of its own, and called `ClientMsqContext` for clarity.